skglm/solvers/group_bcd_solver.py

```diff
@@ -56,51 +59,65 @@ def bcd_solver(X, y, datafit, penalty, w_init=None,
     datafit.initialize(X, y)
     all_groups = np.arange(n_groups)
     p_objs_out = np.zeros(max_iter)
+    stop_crit = 0.  # prevent ref before assign when max_iter == 0
```

Comment:
can you add a check below that penalty is indeed a group penalty (checking it has attribute grp_indices for example) ?

Reply:
I implemented a ``_check_group_compatible`` function for that.

I will add also a unit test for that.

Reply:
no need for a function now, keep it simple with a 2 lines check
@PABannier your thoughts ?

Reply:
I like the `_check_group_compatible` function, although I'd make it more explicit and put it in a utils file instead.
Also make the message error more explicit: `Penalties is not block-separable...`. `Missing attr...` is not explicit enough IMO.

I'd do something like this:
```
def _check_group_compatible(obj):
    group_attrs = set(['grp_ptr', 'grp_indices'])
    obj_attrs = set(dir(obj))
    if not group_attrs.issubset(obj_attrs):
            raise ValueError("Datafits and penalties must be compatible with group solvers")
```

Reply:
`_check_group_compatible` may be reused by other group solvers in the future, while `_bcd_epoch` and `_compute_grad` are tailored for this solver. That's why I advocate to encapsulate the logic in a function and put it in a `utils` module.

For the implementation, a for loop or set operations are both `O(n)` time in the number of attributes of the obj. In practice, penalties and datafits don't even have 100 attributes, so the effective cost remains relatively cheap.
I suggested this approach for conciseness, but I have no strong opinion on that matter.